components/pdf_preview.jsx

```diff
@@ -111,19 +133,35 @@ export default class PDFPreview extends React.PureComponent {
     }
 
     onDocumentLoad = (pdf) => {
-        const numPages = pdf.numPages <= MAX_PDF_PAGES ? pdf.numPages : MAX_PDF_PAGES;
-        this.setState({pdf, numPages});
-        for (let i = 1; i <= pdf.numPages; i++) {
-            pdf.getPage(i).then(this.onPageLoad);
+        this.setState({pdf, numPages: pdf.numPages});
+        for (let i = 0; i < pdf.numPages; i++) {
+            this[`pdfCanvasRef-${i}`] = React.createRef();
         }
+        this.loadPages(pdf);
     }
 
     onDocumentLoadError = (reason) => {
         console.log('Unable to load PDF preview: ' + reason); //eslint-disable-line no-console
         this.setState({loading: false, success: false});
     }
 
-    onPageLoad = (page) => {
+    loadPages = async (pdf) => {
+        const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
```

Comment:
Maybe we could use `requestAnimationFrame` in here instead so that it's throttled by how fast the browser re-renders? That might be a bit better since it'll trigger slower if the user's machine is slower.

Reply:
Very good point.

Reply:
I tried, and doesn't work well, I going to keep my approach because is working correctly and is not hard to understand.